Python/ast.c

```diff
@@ -25,15 +25,15 @@ static int validate_pattern(struct validator *, pattern_ty, int);
 #define VALIDATE_POSITIONS(node) \
     if (node->lineno > node->end_lineno) { \
         PyErr_Format(PyExc_ValueError, \
-                     "line %d-%d is not a valid range", \
+                     "AST node line range (%d, %d) is not valid", \
                      node->lineno, node->end_lineno); \
         return 0; \
     } \
     if ((node->lineno < 0 && node->end_lineno != node->lineno) || \
         (node->col_offset < 0 && node->col_offset != node->end_col_offset)) { \
         PyErr_Format(PyExc_ValueError, \
-                     "line %d-%d, column %d-%d is not a valid range", \
-                     node->lineno, node->end_lineno, node->col_offset, node->end_col_offset); \
+                     "AST node column range (%d, %d) for line %d is not valid", \
```

Comment:
Do we want to include the `end_lineno` here as well (since it is part of the original condition):
```c
    if ((node->lineno < 0 && node->end_lineno != node->lineno) || \
```